### PR TITLE
Implement get serial command

### PR DESCRIPTION
**Pull request by Zifzaf**
Check the following:

 - [na] Breaking changes marked in commit message
 - [x] Changelog updated
 - [x] Code style cleaned (ran `make style-fix`)
 - [x] Tested on actual hardware

### CODE REVIEW

File: svm40/svm40.c [hunk 1]
```diff
@@ -36,6 +36,24 @@ const char* svm40_get_driver_version(void) {
     return SVM40_DRV_VERSION_STR;
 }
 
+int16_t svm40_get_serial(char* serial) {
+    int16_t error;
```
Comment (abrauchli) [inline svm40/svm40.c:40]: I'd recommend a blank line between declaration block and implementation - we don't do it universally but I try to do it.

File: svm40/svm40.c [hunk 2]
```diff
@@ -36,6 +36,24 @@ const char* svm40_get_driver_version(void) {
...
+    error =
+        sensirion_i2c_write_cmd(SVM40_I2C_ADDRESS, SVM40_CMD_GET_SERIAL_NUMBER);
+    if (error != NO_ERROR) {
+        return error;
+    }
+    sensirion_sleep_usec(1);
```
Comment (abrauchli) [inline svm40/svm40.c:46]: Be prepared to answer the "Where is this documented?" question with timings.. Could we make it a constant?

File: svm40/svm40.c [hunk 3]
```diff
@@ -36,6 +36,24 @@ const char* svm40_get_driver_version(void) {
...
+    error = sensirion_i2c_read_words_as_bytes(
+        SVM40_I2C_ADDRESS, (uint8_t*)serial,
+        SVM40_MAX_SERIAL_LEN / SENSIRION_WORD_SIZE);
```
Comment (abrauchli) [inline svm40/svm40.c:49]: have a look at the `SENSIRION_NUM_WORDS()` macro

File: svm40/svm40.h [hunk 4]
```diff
@@ -88,6 +88,16 @@ struct svm40_version_information {
 int16_t
 svm40_get_version(struct svm40_version_information* version_information);
 
+/*
```
Comment (abrauchli) [inline svm40/svm40.h:91]: use `/**` to initiate a docstr comment - i.e. a comment that could be used for automated processing such as for generated documentation.

File: svm40/svm40.h [hunk 5]
```diff
@@ -88,6 +88,16 @@ struct svm40_version_information {
...
+ * Get Serial Number.
+ *
+ * @param serial Memory where the serial number is written into
+ * 				 (zero terminated). Musit be at least
```
Comment (abrauchli) [inline svm40/svm40.h:95]:
```suggestion
 * 				 The string is guaranteed to be zero terminated. It must be at least
```

File: svm40/svm40_example_usage.c [hunk 6]
```diff
@@ -56,8 +57,18 @@ int main(void) {
     }
     printf("SVM40 sensor probing successful\n");
 
+    char serial_id[SVM40_MAX_SERIAL_LEN];
```
Comment (abrauchli) [inline svm40/svm40_example_usage.c:60]: I'd declare it at top of the function (this was mandatory for C89, which don't have to be compatible with, but I still like it :) )

**Comment by rnestler** [comment]
From one of the commit messages:

> - Fix error variable in sample usage to be initialised at the beginning
  of main.

That's not what you did: You changed it to be *declared* at the beginning of main.

File: svm40/svm40_example_usage.c [hunk 7]
```diff
@@ -56,8 +57,18 @@ int main(void) {
...
+    error = svm40_get_serial(serial_id);
+    if (error == NO_ERROR) {
```
Comment (abrauchli) [inline svm40/svm40_example_usage.c:62]: There's not output on error. I generally prefer for the error handling case to be first and the no-error flow to be after:
e.g.

```cpp
if (error != NO_ERROR) {
    printf("Error reading SVM40 serial: %i\n", error);
} else {
    // ...
}
```

File: svm40/svm40_example_usage.c [hunk 8]
```diff
@@ -56,8 +57,18 @@ int main(void) {
...
+        printf("Serial Number: ");
+        for (uint16_t i = 0; i < SVM40_MAX_SERIAL_LEN - 2; i++) {
```
Comment (abrauchli) [inline svm40/svm40_example_usage.c:64]: I don't know if we standardize this, again coming from C89 style, I prefer the declaration of `i` at the top of the function... @rnestler ?

**Review by abrauchli — CHANGES_REQUESTED** [review]

File: svm40/svm40.c (hunk 1 above)
Comment (Zifzaf) [reply svm40/svm40.c:40]: Okay noted :)

File: svm40/svm40.c (hunk 2 above)
Comment (Zifzaf) [reply svm40/svm40.c:46]: Changed to constant.

File: svm40/svm40.c [hunk 9]
```diff
@@ -36,6 +36,25 @@ const char* svm40_get_driver_version(void) {
...
 
+int16_t svm40_get_serial(char* serial) {
+    int16_t error;
+
+    error =
+        sensirion_i2c_write_cmd(SVM40_I2C_ADDRESS, SVM40_CMD_GET_SERIAL_NUMBER);
+    if (error != NO_ERROR) {
+        return error;
+    }
+    sensirion_sleep_usec(SVM40_DELAY_GET_SERIAL_NUMBER);
```
Comment (rnestler) [inline svm40/svm40.c:47]: Is this delay really needed?

File: svm40/svm40_example_usage.c [hunk 10]
```diff
@@ -56,8 +60,19 @@ int main(void) {
     }
     printf("SVM40 sensor probing successful\n");
 
+    error = svm40_get_serial(serial_id);
+    if (error != NO_ERROR) {
+        printf("Error reading SVM40 serial: %i\n", error);
+    } else {
+        printf("Serial Number: ");
+        for (i = 0; i < SVM40_MAX_SERIAL_LEN - 2; i++) {
+            printf("%c", serial_id[i]);
```
Comment (rnestler) [inline svm40/svm40_example_usage.c:69]: Just use `%s` in the format string instead of the for loop and printing each character by its own.

**Review by rnestler — CHANGES_REQUESTED** [review]

File: svm40/svm40_example_usage.c (hunk 10 above)
Comment (abrauchli) [reply svm40/svm40_example_usage.c:69]: `printf("Serial Number: %s", serial_id);` - sorry, I should have seen that, too 🐤

File: svm40/svm40.c (hunk 9 above)
Comment (Zifzaf) [reply svm40/svm40.c:47]: https://rd-hw.gitlab.sensirion.lokal/-/Phoenix/phoenix2-firmware/-/jobs/327019/artifacts/documentation/i2c/SVM40_I2cInterface.html

Yes it's in the specification as read delay of one millisecond!

File: svm40/svm40_example_usage.c [hunk 11]
```diff
@@ -39,6 +39,10 @@
  */
 
 int main(void) {
+    int16_t error;
+    char serial_id[SVM40_MAX_SERIAL_LEN];
+    uint16_t i;
```
Comment (rnestler) [inline svm40/svm40_example_usage.c:44]: This seems like an unused variable, no?

Comment (Zifzaf) [reply svm40/svm40_example_usage.c:44]: Yes now it is, will fix it.

File: svm40/svm40.c (hunk 9 above)
Comment (rnestler) [reply svm40/svm40.c:47]: So we should add it to all the commands then.

Comment (rnestler) [reply svm40/svm40.c:47]: Maybe `#define` a single `SVM40_READ_DELAY` instead of having 4 different defines with the same value.

**Review by rnestler — APPROVED** [review]

**Review by abrauchli — APPROVED** [review]

**Pull request merged**